### PR TITLE
Verificator initial refactor

### PR DESCRIPTION
**Pull request by rwrzesien**
Resolves https://codepoets.atlassian.net/browse/GF-2
Resolves https://codepoets.atlassian.net/browse/GF-3

### CODE REVIEW

File: golem/verificator/core_verifier.py [hunk 1]
```diff
@@ -44,9 +55,53 @@ def verification_completed(self):
...
+        self.time_ended = None
+        self.extra_data = {}
+
+    def _get_answer(self):
+        return {'message': self.message,
+                'time_started': self.time_started,
+                'time_ended': self.time_ended,
+                'extra_data': self.extra_data}
+
+    def _check_computer(self):
```
Comment (dybi) [inline golem/verificator/core_verifier.py]: please remove as it is a dead code - not used anywhere

File: golem/verificator/core_verifier.py [hunk 2]
```diff
@@ -44,9 +55,53 @@ def verification_completed(self):
...
+        if not self.computer:
+            self.state = SubtaskVerificationState.NOT_SURE
+            self.message = 'No computer available to verify data'
+            return False
+        return True
+
+    def _wait_for_computer(self):
```
Comment (dybi) [inline golem/verificator/core_verifier.py]: please remove as it is a dead code - not used anywhere

File: golem/verificator/imgrepr.py [hunk 3]
```diff
@@ -1,19 +1,20 @@
...
 from copy import deepcopy
 from typing import Optional
 
 import OpenEXR
 import Imath
 from PIL import Image
 
-logger = logging.getLogger("apps.rendering")
+
+logger = logging.getLogger('apps.rendering')
```
Comment (dybi) [inline golem/verificator/imgrepr.py]: this file `imagerepr.py` should be removed; exactly the same file is placed in `apps.rendering.resources` and it should be used

File: golem/verificator/imgverifier.py [hunk 4]
```diff
@@ -1,17 +1,19 @@
...
+from random import uniform
+from typing import Optional
 
-from .imgrepr import (ImgRepr, PILImgRepr)
+import numpy
+from ssim import compute_ssim
 
-from .verifier import SubtaskVerificationState
+from golem.verificator.constants import SubtaskVerificationState
+from golem.verificator.imgrepr import (ImgRepr, PILImgRepr)
```
Comment (dybi) [inline golem/verificator/imgverifier.py]: please use this file from different location and remove `golem.verificator.imgrepr`
```suggestion
from apps.rendering.resources import (ImgRepr, PILImgRepr)
```

File: golem/verificator/imgverifier.py [hunk 5]
```diff
@@ -24,21 +26,20 @@ def __init__(self, base_img: ImgRepr, img: ImgRepr):
             self._calculate_greyscale_normalized_mse(base_img, self.img)
         self.psnr = self._calculate_psnr(self.mse)
 
-    @property
-    def name(self) -> str:
+    def name(self) -> Optional[str]:
```
Comment (dybi) [inline golem/verificator/imgverifier.py:30]: why `@property` annotation has been removed?

File: golem/verificator/rendering_verifier.py [hunk 6]
```diff
@@ -22,8 +24,8 @@ def check_size(file_, res_x, res_y):
             return False
         img_x, img_y = img.get_size()
         if img_x != res_x:
-            logger.info("Subtask size doesn't match, has %r,"
-                        " should be %r", img.get_size(), (res_x, res_y))
+            logger.info('Subtask size doesn\'t match, has %r,'
```
Comment (dybi) [inline golem/verificator/rendering_verifier.py]: I would leave it as is (with `""`), not to introduce escape character in the string...

**Review by dybi — CHANGES_REQUESTED** [review]
There are some improvements to be made before merging this. Apart from explicitly highlighted lines, a few general remarks below: 

0. Please change the destination branch to `verifier-cleanup/master`
1. Name abbreviations were to be replaced with full names: `img` -> `image`, etc.
2. `test_verifier` doesn't make sense - there is no `Verifier` class anymore; besides this test is essentially doing nothing - it should be changed (I have done it already os some test branch - will provide the name later). `#TODO`: move contents of `test_verifier` into `test_coreverifier` and remove the former.

EDIT:
One more thing - `lint.sh` fails:
```
golem/verificator/imgverifier.py:64:81: E501 line too long (87 > 80 characters)\n
```

**Comment by kbeker** [comment]
I have test it locally and everything still works :)

File: golem/verificator/core_verifier.py (hunk 1 above)
Comment (rwrzesien) [reply golem/verificator/core_verifier.py]: :+1: done

File: golem/verificator/core_verifier.py (hunk 2 above)
Comment (rwrzesien) [reply golem/verificator/core_verifier.py]: :+1: done

File: golem/verificator/imgrepr.py (hunk 3 above)
Comment (rwrzesien) [reply golem/verificator/imgrepr.py]: :+1: done

File: golem/verificator/imgverifier.py (hunk 4 above)
Comment (rwrzesien) [reply golem/verificator/imgverifier.py]: :+1: done

File: golem/verificator/rendering_verifier.py (hunk 6 above)
Comment (rwrzesien) [reply golem/verificator/rendering_verifier.py]: :+1: done

File: golem/verificator/imgverifier.py (hunk 5 above)
Comment (rwrzesien) [reply golem/verificator/imgverifier.py:30]: Must be by accident, reverted :+1: done

**Comment by rwrzesien** [comment]
@dybi 

> There are some improvements to be made before merging this. Apart from explicitly highlighted lines, a few general remarks below:

All of them are done now.

File: tests/golem/verificator/test_core_verifier.py [hunk 7]
```diff
@@ -51,3 +53,19 @@ def test_simple_verification(self):
         verification_data["results"] = ["not a file"]
         core_verifier.simple_verification(verification_data)
         assert core_verifier.state == SubtaskVerificationState.WRONG_ANSWER
+
+    @staticmethod
+    def test_task_timeout():
```
Comment (dybi) [inline tests/golem/verificator/test_core_verifier.py]: just one thing, please add `#TODO: fix it`, not to forget (I can do it as a part of my current assignment)
```suggestion
    def test_task_timeout():  #TODO: fix it`
```

**Review by dybi — APPROVED** [review]
professor_ok

**Comment by dybi** [comment]
Final thing: `lint.sh` again:
```
...
New findings:

golem/verificator/imgverifier.py:65:81: E501 line too long (93 > 80 characters)


Changed lines findings:

golem/verificator/imgverifier.py:65:81: E501 line too long (93 > 80 characters)\n
-----------------------------------------------------------------------------------------------------------------------------------------------
Errors occurred, summary:
pylint main...       [FAIL]
pylint tests...      [OK]
pycodestyle...       [FAIL]
mypy...              [OK]
```

File: tests/golem/verificator/test_core_verifier.py (hunk 7 above)
Comment (rwrzesien) [reply tests/golem/verificator/test_core_verifier.py]: :+1: done

**Pull request merged**